addons/mass_mailing/models/mail_mail.py

```diff
@@ -5,78 +5,70 @@
 import urlparse
 import werkzeug.urls
 
-from openerp import tools
-from openerp import SUPERUSER_ID
-from openerp.osv import osv, fields
+from odoo import api, fields, models, tools
 
 
 URL_REGEX = r'(\bhref=[\'"]([^\'"]+)[\'"])'
 
 
-class MailMail(osv.Model):
+class MailMail(models.Model):
     """Add the mass mailing campaign data to mail"""
-    _name = 'mail.mail'
     _inherit = ['mail.mail']
 
-    _columns = {
-        'mailing_id': fields.many2one('mail.mass_mailing', 'Mass Mailing'),
-        'statistics_ids': fields.one2many(
-            'mail.mail.statistics', 'mail_mail_id',
-            string='Statistics',
-        ),
-    }
+    mailing_id = fields.Many2one('mail.mass_mailing', string='Mass Mailing')
+    statistics_ids = fields.One2many('mail.mail.statistics', 'mail_mail_id', string='Statistics')
 
-    def create(self, cr, uid, values, context=None):
+    @api.model
+    def create(self, values):
         """ Override mail_mail creation to create an entry in mail.mail.statistics """
         # TDE note: should be after 'all values computed', to have values (FIXME after merging other branch holding create refactoring)
-        mail_id = super(MailMail, self).create(cr, uid, values, context=context)
+        mail = super(MailMail, self).create(values)
         if values.get('statistics_ids'):
-            mail = self.browse(cr, SUPERUSER_ID, mail_id, context=context)
-            for stat in mail.statistics_ids:
-                self.pool['mail.mail.statistics'].write(cr, uid, [stat.id], {'message_id': mail.message_id, 'state': 'outgoing'}, context=context)
-        return mail_id
+            mail.statistics_ids.write({'message_id': mail.message_id, 'state': 'outgoing'})
+        return mail
 
-    def _get_tracking_url(self, cr, uid, mail, partner=None, context=None):
-        base_url = self.pool.get('ir.config_parameter').get_param(cr, uid, 'web.base.url')
+    def _get_tracking_url(self, partner=None):
```

Comment:
xmo would say "don't add it if it's working without", or something more philosophical. He prefer remove unrelevant/redundant decorator. So I don't force you to add it, except if it causes problems ;)